tests/validation/tests/v3_api/test_cluster_templates.py

```diff
@@ -0,0 +1,448 @@
+
+    cluster_config1 = get_cluster_config(RANCHER_DEFAULT_K8S_VERSION)
+    client = get_admin_client()
+
+    standard_newuser_client = Client(
+                                url=CATTLE_API_URL,
+                                token=user_token["newuser_standard"]["token"],
+                                verify=False)
+    # As an Admin, create a cluster template, revision and update the members
+    # list with the new user as read-only user
+    template_name = random_test_name("usertemplate")
+    cluster_template = client.create_cluster_template(
+        name=template_name, description="test-template")
+
+    revision_name = random_test_name("revision1")
+    cluster_template_revision1 = client.create_cluster_template_revision(
+        name=revision_name,
+        clusterConfig=cluster_config1,
+        clusterTemplateId=cluster_template.id)
+
+    principalid = user_token["newuser_standard"]["user"]["principalIds"]
+    members = [{
+        "type": "member",
+        "accessType": "read-only",
+        "userPrincipalId": principalid
+    }]
+
+    cluster_template = client.update(cluster_template,
+                                     name=template_name, members=members)
+
+    # As a read-only member of the rke template verify that
+    # adding another revision to the template fails
+    revision_name = "userrevision"
+    with pytest.raises(ApiError) as e:
+        standard_newuser_client.create_cluster_template_revision(
+            name=revision_name,
+            clusterConfig=cluster_config1,
+            clusterTemplateId=cluster_template.id)
+
+    assert e.value.error.status == 403
+    assert e.value.error.code == 'PermissionDenied'
+
+    # Verify that the read-only user can create a cluster with the existing
+    # template revision
+    cluster_name = "test-auto"
+    cluster = create_node_cluster(
+        standard_newuser_client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision1.id)
+
+    cluster_cleanup(standard_newuser_client, cluster)
+
+
+def cluster_template_create_edit(testclient):
+
+    cluster_config1 = get_cluster_config(RANCHER_DEFAULT_K8S_VERSION)
+    cluster_config2 = get_cluster_config("v1.15.4-rancher1-2")
+
+    client = testclient
+
+    cluster_template = client.create_cluster_template(
+        name=random_test_name("template"), description="test-template")
+
+    cluster_template_revision1 = client.create_cluster_template_revision(
+        name=random_test_name("revision1"),
+        clusterConfig=cluster_config1,
+        clusterTemplateId=cluster_template.id)
+
+    cluster_name = random_test_name("test-auto")
+    cluster = create_node_cluster(
+        client, name=cluster_name,
+        clusterTemplateRevisionId=cluster_template_revision1.id)
+
+    cluster_template_revision2 = client.create_cluster_template_revision(
+        name=random_test_name("revision2"),
+        clusterConfig=cluster_config2,
+        clusterTemplateId=cluster_template.id)
+
```

Comment:
Assert k8s versions of the cluster using kubectl command

Reply:
check_cluster_version(cluster,version) method in common.py